### ISSUE TITLE
Remember size and position for preferences and config editor dialogs

### ISSUE THREAD

**Issue by auouymous** [gpodder/gpodder#704]
The position and size of main window and some other dialogs are remembered, but that doesn't happen for the preferences and edit config dialogs. My gpodder window is narrower than the preferences dialog and always sits on the right edge of the screen. The default behavior of centering the preferences dialog over the gpodder window causes the right side to be off the edge of the screen.

```
diff --git a/src/gpodder/config.py b/src/gpodder/config.py
index cc7c970f..d876d90d 100644
--- a/src/gpodder/config.py
+++ b/src/gpodder/config.py
@@ -115,6 +115,16 @@ defaults = {
                     'paned_position': 200,
                     'episode_list_size': 200,
                 },
+                'preferences': {
+                    'width': -1,
+                    'height': -1,
+                    'x': -1, 'y': -1, 'maximized': False,
+                },
+                'config_editor': {
+                    'width': -1,
+                    'height': -1,
+                    'x': -1, 'y': -1, 'maximized': False,
+                },
                 'episode_selector': {
                     'width': 600,
                     'height': 400,
diff --git a/src/gpodder/gtkui/config.py b/src/gpodder/gtkui/config.py
index 9cfbc4f3..7168aef0 100644
--- a/src/gpodder/gtkui/config.py
+++ b/src/gpodder/gtkui/config.py
@@ -156,7 +156,8 @@ class UIConfig(config.Config):
         if gpodder.ui.win32:
             window.set_gravity(Gdk.Gravity.STATIC)
 
-        window.resize(cfg.width, cfg.height)
+        if cfg.width != -1 and cfg.height != -1:
+            window.resize(cfg.width, cfg.height)
         if cfg.x == -1 or cfg.y == -1:
             window.set_position(Gtk.WindowPosition.CENTER_ON_PARENT)
         else:
diff --git a/src/gpodder/gtkui/desktop/preferences.py b/src/gpodder/gtkui/desktop/preferences.py
index e7794c6e..23efae11 100644
--- a/src/gpodder/gtkui/desktop/preferences.py
+++ b/src/gpodder/gtkui/desktop/preferences.py
@@ -287,7 +287,8 @@ class gPodderPreferences(BuilderWidget):
 
         # Configure the extensions manager GUI
         self.set_extension_preferences()
-        self.main_window.show()
+
+        self._config.connect_gtk_window(self.main_window, 'preferences', True)
 
     def _extensions_select_function(self, selection, model, path, path_currently_selected):
         return model.get_value(model.get_iter(path), self.C_SHOW_TOGGLE)
diff --git a/src/gpodder/gtkui/interface/configeditor.py b/src/gpodder/gtkui/interface/configeditor.py
index 8ae3a9fc..02c73a08 100644
--- a/src/gpodder/gtkui/interface/configeditor.py
+++ b/src/gpodder/gtkui/interface/configeditor.py
@@ -65,6 +65,8 @@ class gPodderConfigEditor(BuilderWidget):
         self.configeditor.set_model(self.filter)
         self.configeditor.set_rules_hint(True)
 
+        self._config.connect_gtk_window(self.main_window, 'config_editor', True)
+
     def visible_func(self, model, iter, user_data=None):
         text = self.entryFilter.get_text().lower()
         if text == '':
```

**Comment by elelay** [comment]
Hi @auouymous, your valuable contributions are very welcome, but please open a PR for longer fixes, like this one.

**Comment by auouymous** [comment]
Is this method acceptable?

```
wget http://borg.qzx.com/files/gpodder-704-save-position-and-size-for-preferences-and-editcfg-windows.patch
git am gpodder-704-save-position-and-size-for-preferences-and-editcfg-windows.patch
rm gpodder-704-save-position-and-size-for-preferences-and-editcfg-windows.patch
```

**Comment by elelay** [comment]
closed by a46f131785fbb3bb34f456e25d60c99b3f19b8f7
